geth/whisper/request_historic_messages.go

```diff
@@ -26,27 +27,25 @@ var (
 	//ErrMailboxSymkeyIDNotString - error symKeyID is not string type
 	ErrMailboxSymkeyIDNotString = fmt.Errorf("symKeyID is not string")
 	//ErrPeerNotExist - error peer field doesn't exist in request
-	ErrPeerNotExist = fmt.Errorf("peer does not exist")
-	//ErrPeerNotString - error peer is not string type
-	ErrPeerNotString = fmt.Errorf("peer is not string")
+	ErrPeerOrEnode = fmt.Errorf("enode or peer field should be not empty")
 )
 
 const defaultWorkTime = 5
 
 //RequestHistoricMessagesHandler returns an RPC handler which sends a p2p request for historic messages.
-func RequestHistoricMessagesHandler(nodeManager common.NodeManager) (rpc.Handler, error) {
-	whisper, err := nodeManager.WhisperService()
-	if err != nil {
-		return nil, err
-	}
+func RequestHistoricMessagesHandler(nodeManager common.NodeManager) rpc.Handler {
+	return func(ctx context.Context, args ...interface{}) (interface{}, error) {
+		whisper, err := nodeManager.WhisperService()
+		if err != nil {
+			return nil, err
+		}
 
-	node, err := nodeManager.Node()
-	if err != nil {
-		return nil, err
-	}
+		node, err := nodeManager.Node()
+		if err != nil {
+			return nil, err
+		}
 
-	return func(ctx context.Context, args ...interface{}) (interface{}, error) {
-		r, err := parseArgs(args)
+		r, err := parseArgs(args...)
```

Comment:
You could change `func parseArgs(args []interface{})` to avoid `...` everywhere.

Reply:
I think to do this i have to change  Handler interface `type Handler func(context.Context, ...interface{}) (interface{}, error)`
It should be separated task like rpc refactoring.

Reply:
No, I am talking about `parseArgs` only. You call it like `parseArgs(args...)` and its signature is `func parseArgs(args ...interface{})`. It's the same as `parseArgs(args)` and `func parseArgs(args []interface{})`. It's a detail, not important now.